src/openms/include/OpenMS/KERNEL/MSSpectrum.h

```diff
@@ -330,6 +330,78 @@ namespace OpenMS
       return integer_data_arrays_;
     }
 
+    /// Returns an iterator to the first float data array with the given name. 
+    /// The End iterator is returned in case no data array with given name exists.
+    typename FloatDataArrays::iterator getFloatDataArrayByName(const String& name)
```

Comment:
I wonder how useful it is to return a start iterator but have no way to access the end iterator? Wouldnt it be easier to return a reference to the whole array and let the user decide how to iterate?